src/str.js

```diff
@@ -599,39 +719,7 @@ Sk.builtin.str.prototype["index"] = new Sk.builtin.func(function (self, tgt, sta
     return idx;
 });
 
-Sk.builtin.str.prototype["rfind"] = new Sk.builtin.func(function (self, tgt, start, end) {
-    var idx;
-    Sk.builtin.pyCheckArgsLen("rfind", arguments.length, 2, 4);
-    if (!Sk.builtin.checkString(tgt)) {
-        throw new Sk.builtin.TypeError("expected a character buffer object");
-    }
-    if ((start !== undefined) && !Sk.builtin.checkInt(start) && !Sk.builtin.checkNone(start)) {
-        throw new Sk.builtin.TypeError("slice indices must be integers or None or have an __index__ method");
-    }
-    if ((end !== undefined) && !Sk.builtin.checkInt(end) && !Sk.builtin.checkNone(end)) {
-        throw new Sk.builtin.TypeError("slice indices must be integers or None or have an __index__ method");
-    }
-
-    if (start === undefined || Sk.builtin.checkNone(start)) {
-        start = 0;
-    } else {
-        start = Sk.builtin.asnum$(start);
-        start = start >= 0 ? start : self.v.length + start;
-    }
-
-    if (end === undefined || Sk.builtin.checkNone(end)) {
-        end = self.v.length;
-    } else {
-        end = Sk.builtin.asnum$(end);
-        end = end >= 0 ? end : self.v.length + end;
-    }
-
-    idx = self.v.lastIndexOf(tgt.v, end);
-    idx = (idx !== end) ? idx : self.v.lastIndexOf(tgt.v, end - 1);
-    idx = ((idx >= start) && (idx < end)) ? idx : -1;
-
-    return new Sk.builtin.int_(idx);
-});
+Sk.builtin.str.prototype["rfind"] = mkFind(true);
 
 Sk.builtin.str.prototype["rindex"] = new Sk.builtin.func(function (self, tgt, start, end) {
     var idx;
```

Comment:
*(A few lines down from here, but GH won't let me put the comment there...)*

`startswith()` and `endswith()` now need to be astral-aware too!

Reply:
have done this and extracted the indices function used by `mkFind`, `startswith`, `endswith` you might want to check this in s-cork#10